.github/workflows/test-and-publish-nightly.yml

```diff
@@ -94,41 +92,28 @@ jobs:
               make COMMIT_TAG=$HASH CC=/usr/local/musl/bin/musl-gcc coverage -j
             ;;
           esac
-      - name: Upload test logs for default architecture
+      - name: Upload test logs
         uses: actions/upload-artifact@v4
 
         if: always() # we always want to upload test logs, especially when tests fail
         with:
-          name: test-logs-${{ matrix.runson.display }}-${{ matrix.java-version }}
+          name: test-logs-${{ matrix.runson.display }}-${{ matrix.java-version }}${{ matrix.architecture && '-' || '' }}${{ matrix.architecture }}
           path: |
             build/test/logs/
             hs_err*.log
       - name: Upload coverage report
         uses: actions/upload-artifact@v4
 
         with:
-          name: test-coverage-${{ matrix.runson.display }}-${{ matrix.java-version }}
+          name: test-coverage-${{ matrix.runson.display }}-${{ matrix.java-version }}${{ matrix.architecture && '-' || '' }}${{ matrix.architecture }}
           path: build/test/coverage/
-      - name: Test macOS x64
-        if: ${{ matrix.runson.name == 'macos-14' }}
-        run: JAVA_HOME=$JAVA_HOME_${{ env.JAVA_VERSION }}_X64 make test
-
       - name: Upload async-profiler binaries to workflow
         uses: actions/upload-artifact@v4
 
-        if: env.RELEASE_FROM_JAVA_VERSION == matrix.java-version
+        if: ${{ matrix.java-version == '11' && matrix.runson.name != 'macos-13' }}
```

Comment:
This will exclude macOS from releases entirely.
I suggest we release macOS binaries from builds on `macos-14`.

Reply:
Should we release only binaries built with JDK 11? `macos-14` now uses JDK 21 due to [this](https://github.com/async-profiler/async-profiler/pull/1212#discussion_r2027448323) previous comment

Reply:
Please check now: https://github.com/async-profiler/async-profiler/actions/runs/14266840749